File: frontend/src/types/announcement.ts
export interface AnnouncementSummaryDto {
  id: string
  siteId: string
  title: string
  isPinned: boolean
  publishDate?: string
  expiryDate?: string
  createdAt: string
}

export interface AnnouncementDetailDto extends AnnouncementSummaryDto {
  createdByUserId: string
  content: string
  updatedAt?: string
}

export interface CreateAnnouncementDto {
  title: string
  content: string
  isPinned: boolean
  publishDate?: string
  expiryDate?: string
}

export interface UpdateAnnouncementDto {
  title: string
  content: string
  isPinned: boolean
  publishDate?: string
  expiryDate?: string
}

File: backend/src/SiteYonetimi.Infrastructure/Entities/Shared/Announcement.cs
namespace SiteYonetimi.Infrastructure.Entities.Shared;

public class Announcement
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid SiteId { get; set; }
    public Guid CreatedByUserId { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public bool IsPinned { get; set; } = false;
    public DateTime? PublishDate { get; set; }
    public DateTime? ExpiryDate { get; set; }
    public bool IsDeleted { get; set; } = false;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
}
